quacc/calculators/qchem.py

```diff
@@ -0,0 +1,132 @@
+
+
+class QChem(FileIOCalculator):
+    """
+
+    Parameters
+    ----------
+    input_atoms
+        The Atoms object to be used for the calculation.
+    use_custodian
+        Whether to use Custodian to run Q-Chem.
+        Default is True in settings.
+    **kwargs
+        Additional arguments to be passed to the Q-Chem calculator. Takes all valid
+        ASE calculator arguments, in addition to those custom to Quacc.
+
+    Returns
+    -------
+    Atoms
+        The ASE Atoms object with attached Q-Chem calculator.
+    """
+
+    def __init__(
+        self,
+        input_atoms: Atoms,
+        charge: None | int = None,
+        spin_multiplicity: None | int = None,
+        qchem_input_params: dict = None,
+        use_custodian: bool = SETTINGS.QCHEM_CUSTODIAN,
+        **kwargs,
+    ):
+        # Assign variables to self
+        self.input_atoms = input_atoms
+        self.charge = charge
+        self.spin_multiplicity = spin_multiplicity
+        self.qchem_input_params = qchem_input_params
+        self.use_custodian = use_custodian
+        self.qchem_input_params = qchem_input_params or {}
+        self.kwargs = kwargs
+
+        # Instantiate the calculator
+        FileIOCalculator.__init__(
+            self,
+            restart=None,
+            ignore_bad_restart_file=FileIOCalculator._deprecated,
+            label=None,
+            atoms=self.input_atoms,
+            **self.kwargs,
+        )
+
+        # Get Q-Chem executable command
+        command = self._manage_environment()
+
+    def _manage_environment(self) -> str:
+        """
+        Manage the environment for the Q-Chem calculator.
+
+        Returns
+        -------
+        str
+            The command flag to pass to the Q-Chem calculator.
+        """
+
+        # Check if Custodian should be used
+        if self.use_custodian:
+            # Return the command flag
+            run_qchem_custodian_file = os.path.abspath(inspect.getfile(custodian_qchem))
+            return f"python {run_qchem_custodian_file}"
+        else:
+            warnings.warn("Can only run Q-Chem via Custodian currently!")
+            return None
+
+    def write_input(self, atoms, properties=None, system_changes=None):
+        FileIOCalculator.write_input(self, atoms, properties, system_changes)
+        mol = AseAtomsAdaptor.get_molecule(atoms)
+        if self.charge is not None:
+            if self.spin_multiplicity is not None:
+                mol.set_charge_and_spin(
+                    charge=self.charge, spin_multiplicity=self.spin_multiplicity
+                )
+            else:
+                mol.set_charge_and_spin(charge=self.charge)
+        qcin = ForceSet(mol, **self.qchem_input_params)
+        qcin.write("mol.qin")
+
+    def read_results(self):
+        data = QCOutput("mol.qout").data
+        self.results["energy"] = data["final_energy"] * units.Hartree
+        tmp_grad_data = []
+        with zopen("131.0", mode="rb") as file:
```

Comment:
FYI @arosen93, was just thinking about this more and realized this file will only be present if we are using Custodian. In fact, the entire approach that I'm taking only makes sense if we're using Custodian. Can we just hard-code that you must use Custodian rather than giving the user the choice not to?

Reply:
I was thinking about this as well. In an ideal world, I would really like to allow the calculator to operate like either a "normal" ASE calculator where it just calls the executable as well as with Custodian (similar to what is done with the VASP calculator). There are multiple philosophical reasons for this that I won't bore you with. However, if it really does not end up being practical or feasible with the setup you have going, I would entertain the idea of having a custom Q-chem custodian calculator, which would ultimately need to be reflected in the name of the class and such.